src/palace.pyx

```diff
@@ -431,6 +438,14 @@ cdef class Context:
         """Update the context and all sources belonging to this context."""
         self.impl.update()
 
+    @staticmethod
+    def current_context() -> Optional[Context]:
```

Comment:
> I make current_context to be a static method instead of make it a module-level function

All you have to do is to move this out of the Context scope.

Reply:
I thought functions outside a class cannot access private variables?

Reply:
There's no such thing as private in Python.  It's a language for adult.  Underscore only hide it from the docs.

BTW this should fetch from alure C++ API so...

Reply:
> There's no such thing as private in Python.

If I add `__` then it will raise an `AttributeError` though.

Reply:
It's still accessible, just differently (`Context._Context__context_stack`).  [Name mangling](https://docs.python.org/3/reference/expressions.html#atom-identifiers) is really to avoid subclasses from accidentally overriding a method that is used by other, e.g.
```python
class Foo:
     def bar(self): return 42000
     def baz(self): return self.__bar() + 69
     __bar = bar
```
now if it is subclassed as follows
```python
class Phoo(Foo):
     def bar(self): return 420
     __bar = bar
```
`Foo().baz()` is still 42069, since `Foo.__bar` is saved (mangled) as `Foo._Foo__bar` and `Phoo.__bar` is mangled to `Phoo._Phoo__bar`, so despite extreme-late binding, `baz` still calls the super variant.  This has absolutely nothing to do with or case here tho.

Reply:
gentle ping